snuba/pipeline/single_query_plan_pipeline.py

```diff
@@ -35,24 +64,31 @@ def __init__(
         self.__query_plan_builder = query_plan_builder
 
     def execute(self) -> QueryResult:
-        # Execute entity processors
-        _execute_entity_processors(self.__request)
-
-        # Build and execute query plan
-        self.__query_plan = self.__query_plan_builder.build_plan(self.__request)
-        _execute_query_plan_processors(self.__query_plan, self.__request)
-        return self.__query_plan.execution_strategy.execute(
-            self.__query_plan.query, self.__request.settings, self.__runner
-        )
+        settings = self.__request.settings
+        query = self.__request.query
+
+        execute_entity_processors(query, settings)
+
+        query_plan = self.__query_plan_builder.build_plan(query, settings)
+        execute_pre_strategy_processors(query_plan, settings)
```

Comment:
Does this mean we are executing the same query plan processors again? Isn't that already done by `execute_all_clickhouse_processors()` on line 47.

From what I gather, the processing pipeline works is supposed to run before the execution pipeline, though I'm not sure and the processing pipeline doesn't seem to be called anywhere yet. Is there a reason for that? I would have thought we'd still want to run that pipeline (a simplified version at least) for our simple query plans as well (not just relevant to joins)

Reply:
The SunglePlanExecutionPipeline does not execute the ProcessingPipeline. Originally I wanted to make the execution pipeline depend on the processing pipeline, but that turned out to be hard to grasp. See the comment from Evan (https://github.com/getsentry/snuba/pull/1491#discussion_r524384565). 
I did not double down on that route also because there really are three types of processing pipeline thus much less gain in reusing any of them as none of which can be reused.
1) A full processing pipeline. The one in this file. That is basically useful in tests and in the first iteration of composite queries. It executes all clickhouse query processing and skips the strategy.
2) A pre-strategy processing pipeline. Which would be used by the execution pipeline, because it only executes the processor before the strategy, instead of all of them.
3) A pre-plan processing which will be executed in the second iteration of the composite query, which allows building all the plans, merging back to select plans inn the same storage set and fork again.

So in the end we will have three use cases and three pipelines, then at that point, putting the pre strategy processing in its own pipeline when it is used by the execution only seems a lot less useful.